### PR TITLE
Adds NewRelic beta Kubernetes integration

### PR DESCRIPTION
**Pull request by sstarcher**

### CODE REVIEW

File: stable/newrelic-infrastructure/templates/daemonset-k8s.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
+{{- if and .Values.licenseKey .Values.k8s.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels: {{ include "newrelic.k8s.labels" . | indent 4 }}
+  name: {{ template "newrelic.fullname" . }}-k8s
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "newrelic.name" . }}-k8s
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/daemonset-k8s.yaml]: `app: {{ template "newrelic.name" . }}`

File: stable/newrelic-infrastructure/templates/daemonset-k8s.yaml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      {{ toYaml .Values.podLabels }}
+      {{- end }}
+  template:
+    metadata:
+    {{- if .Values.podAnnotations }}
+      annotations:
+{{ toYaml .Values.podAnnotations | indent 8}}
+    {{- end }}
+      labels:
+        app: {{ template "newrelic.name" . }}-k8s
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/daemonset-k8s.yaml]: `app: {{ template "newrelic.name" . }}`

File: stable/newrelic-infrastructure/templates/daemonset-k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+{{- if and .Values.licenseKey .Values.k8s.enabled }}
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  labels: {{ include "newrelic.k8s.labels" . | indent 4 }}
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/daemonset-k8s.yaml]: I'd just keep one partial for labels that does not include the component label and add the component label separately. That's more explicit.

File: stable/newrelic-infrastructure/templates/daemonset-k8s.yaml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  name: {{ template "newrelic.fullname" . }}-k8s
+spec:
+  selector:
+    matchLabels:
+      app: {{ template "newrelic.name" . }}-k8s
+      release: {{.Release.Name }}
+      {{- if .Values.podLabels}}
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/daemonset-k8s.yaml]: I don' think selectors should have additional pod labels. That's unnecessary. However, selector need the component label. Apply everywhere.

File: stable/newrelic-infrastructure/templates/daemonset-k8s.yaml [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+            - name: NRIA_LICENSE_KEY
+              valueFrom:
+                secretKeyRef:
+                  name: {{ template "newrelic.fullname" . }}-config
+                  key: license
+            - name: "CLUSTER_NAME"
+              value: "{{ .Values.k8s.cluster }}"
+            {{- if .Values.verboseLog }}
+            - name: NRIA_VERBOSE
+              value: 1
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/daemonset-k8s.yaml]: You will probably have to quote this: `"1"`

**Review by unguiculus — CHANGES_REQUESTED** [review]

**Comment by sstarcher** [comment]
@unguiculus thanks again for the review

**Comment by unguiculus** [comment]
Next time please don't rebase so Github can display the changes. Thanks.

File: stable/newrelic-infrastructure/templates/rbac.yaml [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{{- if .Values.rbac.create }}
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ template "newrelic.fullname" . }}
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/rbac.yaml]: Add standard labels.

File: stable/newrelic-infrastructure/templates/_helpers.tpl [hunk 7]
```diff
@@ -20,9 +20,37 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
...
+{{- define "newrelic.labels" }}
+app: {{ template "newrelic.name" . }}
 chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+component: infra
 heritage: {{.Release.Service }}
 release: {{.Release.Name }}
 {{- end }}
+
+{{/* Generate basic labels */}}
+{{- define "newrelic.k8s.labels" }}
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/_helpers.tpl]: This partial is obsolete now. Remove it.

File: stable/newrelic-infrastructure/templates/_helpers.tpl [hunk 8]
```diff
@@ -20,9 +20,37 @@ We truncate at 63 chars because some Kubernetes name fields are limited to this
 {{- end -}}
 
 {{/* Generate basic labels */}}
-{{- define "newrelic-infra.labels" }}
-app: {{ template "newrelic-infra.name" . }}
+{{- define "newrelic.labels" }}
+app: {{ template "newrelic.name" . }}
 chart: {{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}
+component: infra
```
Comment (unguiculus) [inline stable/newrelic-infrastructure/templates/_helpers.tpl]: Now everything gets this component label. I wanted you to remove the label here and add the correct component labels explicitly.

**Review by unguiculus — CHANGES_REQUESTED** [review]

File: stable/newrelic-infrastructure/templates/_helpers.tpl (hunk 8 above)
Comment (sstarcher) [reply stable/newrelic-infrastructure/templates/_helpers.tpl]: that slipped through my refactor sorry about that

**Comment by sstarcher** [comment]
@unguiculus thanks again for the review I missed those things.

**Review by unguiculus — APPROVED** [review]

**Pull request merged**